dynawo/sources/API/PAR/PARParametersSetCollection.cpp

```diff
@@ -47,11 +48,40 @@ ParametersSetCollection::addParametersSet(shared_ptr<ParametersSet> paramSet, bo
   parametersSets_[id] = paramSet;
 }
 
+void
+ParametersSetCollection::addMacroParameterSet(shared_ptr<MacroParameterSet> macroParamSet) {
+  assert(macroParamSet && "impossible to add null macro parameter set pointer to collection");
+  string id = macroParamSet->getId();
+  if (hasMacroParametersSet(id)) {
+    throw DYNError(Error::API, MacroParameterSetAlreadyExists, id);
+  }
+  macroParametersSets_[id] = macroParamSet;
+}
+
 shared_ptr<ParametersSet>
 ParametersSetCollection::getParametersSet(const string& id) {
   map< string, shared_ptr<ParametersSet> >::iterator itParamSet = parametersSets_.find(id);
   if (itParamSet == parametersSets_.end())
     throw DYNError(DYN::Error::API, ParametersSetNotFound, id);
+  // if a macroParSet is defined in the set, we add the references and parameters associated
+  if (itParamSet->second->hasMacroParSet()) {
```

Comment:
This will be done everythime we read the parameter even if its used by many models. Shouldn't we factorize this and do it only once?

Reply:
solved separating it in a different method called only once for each par file